src/main/java/hudson/plugins/git/util/BuildData.java

```diff
@@ -243,6 +275,57 @@ public String toString() {
                 ",lastBuild="+lastBuild+"]";
     }
 
+    /**
+     * Like {@link #equals(Object)} but doesn't check the branch names as strictly  as those can vary depending on the
+     * configured remote name.
+     *
+     * @param that the {@link BuildData} to compare with.
+     * @return {@code true} if the supplied {@link BuildData} is similar to this {@link BuildData}.
+     * @since TODO
+     */
+    public boolean similarTo(BuildData that) {
+        if (this.remoteUrls == null ? that.remoteUrls != null : !this.remoteUrls.equals(that.remoteUrls)) {
+            return false;
+        }
+        if (this.lastBuild == null ? that.lastBuild != null : !this.lastBuild.equals(that.lastBuild)) {
+            return false;
+        }
+        if (this.remoteUrls == null || that.remoteUrls == null) {
+            // if either is null then we do not need the costly comparison
+            return this.remoteUrls == that.remoteUrls;
+        }
+        int thisSize = this.remoteUrls.size();
+        int thatSize = that.remoteUrls.size();
+        if (thisSize != thatSize) {
+            return false;
+        }
+        // assume if there is a prefix/ that the prefix is the origin name and strip it for similarity comparison
+        // now if branch names contain slashes anyway and the user has not configured an origin name
+        // we could have a false positive... but come on, it's the same repo and the same revision on the same build
```

Comment:
perhaps, but not relevant to this code... keep in mind that we are on the exact same revision if we even get this far